File: MES_EQUIPMENT/Models/ProductionReportDto.cs
// 백엔드에서 내용 추가해서 넣기

// 생산 결과를 보고하는 DTO (백엔드 서버와 JSON으로 주고 받음)
// DTO에 SerialNo 추가

public class ProductionReportDto
{
    public long WorkOrderId { get; set; }
    public long LotId { get; set; }
    public long MemberId { get; set; }
    public long EquipmentId { get; set; }
    public DateTime StartTime { get; set; }
    public DateTime EndTime { get; set; }
    public string? DefectCode { get; set; }

    public DicingDto? DicingDto { get; set; }
    public DicingInspectionDto? DicingInspectionDto { get; set; }
    public DieBondingDto? DieBondingDto { get; set; }
    public DieBondingInspectionDto? DieBondingInspectionDto { get; set; }
    public WireBondingDto? WireBondingDto { get; set; }
    public WireBondingInspectionDto? WireBondingInspectionDto { get; set; }
    public MoldingDto? MoldingDto { get; set; }
    public MoldingInspectionDto? MoldingInspectionDto { get; set; }
    public ItemDto[]? ItemDtos { get; set; }
    public FinalInspectionDto[]? FinalInspectionDtos { get; set; }
    public string[]? InputLots { get; set; }


}

File: MES_SPRING/src/main/java/com/hm/mes_final_260106/dto/ProductionReportDto.java
package com.hm.mes_final_260106.dto;
// 생산 실적 보고 : 현장에서 생산한 1건의 생산 결과를 시스템에 보고하기 위한 입력 DTO
// 설비 -> 서버

import com.hm.mes_final_260106.entity.LotMapping;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ProductionReportDto {

    // --- 기존 필드 유지 ---
    private Long workOrderId;           // 작업 지시 ID
    private Long lotId;                 // lot ID
    private Long memberId;              // 작업자 ID
    private Long equipmentId;           // 설비 ID
    private LocalDateTime startTime;    // 작업 시작 시간
    private LocalDateTime endTime;      // 작업 시작 시간
    private String defectCode;          // 불량 코드

    // --- 상세 공정 데이터 (Nested DTOs) ---
    // 각 공정 단계별 데이터가 있을 경우에만 채워서 보냄 (Nullable)

    private DicingDto dicingDto;
    private DicingInspectionDto dicingInspectionDto;

    private DieBondingDto dieBondingDto;
    private DieBondingInspectionDto dieBondingInspectionDto;

    private WireBondingDto wireBondingDto;
    private WireBondingInspectionDto wireBondingInspectionDto;

    private MoldingDto moldingDto;
    private MoldingInspectionDto moldingInspectionDto;

    // 배열([]) 대신 List<> 사용 권장
    private List<String> inputLots;
    private List<ItemDto> itemDtos;
    private List<FinalInspectionDto> finalInspectionDtos;
}
